service/frontend/workflowHandler.go

```diff
@@ -3755,3 +3739,81 @@ func (wh *WorkflowHandler) validateRetryPolicy(namespace string, retryPolicy *co
 	common.EnsureRetryPolicyDefaults(retryPolicy, defaultWorkflowRetrySettings)
 	return common.ValidateRetryPolicy(retryPolicy)
 }
+
+func (wh *WorkflowHandler) validateStartWorkflowTimeouts(
+	scope metrics.Scope,
+	request *workflowservice.StartWorkflowExecutionRequest) error {
+	if timestamp.DurationValue(request.GetWorkflowExecutionTimeout()) < 0 {
+		return wh.error(errInvalidWorkflowExecutionTimeoutSeconds, scope)
+	}
+
+	if timestamp.DurationValue(request.GetWorkflowRunTimeout()) < 0 {
+		return wh.error(errInvalidWorkflowRunTimeoutSeconds, scope)
+	}
+
+	if timestamp.DurationValue(request.GetWorkflowTaskTimeout()) < 0 {
+		return wh.error(errInvalidWorkflowTaskTimeoutSeconds, scope)
+	}
+
+	request.WorkflowExecutionTimeout = common.GetWorkflowExecutionTimeout(
```

Comment:
@samarabbas - "the user cannot change the WF timeout values for already running Workflows, which could be problematic (specifically for WorkflowTaskTimeout)."

Do you think this is an issue? Generally if the user is setting these values explicitly, they are already broken, but there could be folks who rely on server defaults and want the flexibility of that to change (if this is a supported scenario)